### PR TITLE
[BEAM-3756] Update SpannerIO to use Batch API.

### PR DESCRIPTION
**Pull request by mairbek**
Migrating SpannerIO to use Batch API.

WIP: waiting for Batch API rollout.

### CODE REVIEW

**Comment by chamikaramj** [comment]
Please let me know when this is ready for review.

cc: @jkff

**Comment by chamikaramj** [comment]
Run Java PostCommit

File: pom.xml [hunk 1]
```diff
@@ -2160,7 +2160,7 @@
             <configuration>
               <!-- Ignore runtime-only dependencies in analysis -->
               <ignoreNonCompile>true</ignoreNonCompile>
-              <failOnWarning>true</failOnWarning>
+              <failOnWarning>false</failOnWarning>
```
Comment (chamikaramj) [inline pom.xml]: Why was this needed. Can we update the jar without disabling 'failOnWarning' for all warnings ?

Comment (mairbek) [reply pom.xml]: For some reason fails locally with

```
[INFO] --- maven-dependency-plugin:3.0.2:analyze-only (default) @ beam-sdks-java-extensions-google-cloud-platform-core ---
[WARNING] Unused declared dependencies found:
[WARNING]    com.fasterxml.jackson.core:jackson-databind:jar:2.8.9:compile
```

reverted

**Comment by chamikaramj** [comment]
LGTM.

Will merge after tests pass.

**Comment by chamikaramj** [comment]
Run Java PostCommit

**Comment by mairbek** [comment]
Run Java PostCommit

**Comment by chamikaramj** [comment]
Run Java Gradle PreCommit

**Comment by chamikaramj** [comment]
Successful post-commit run: https://builds.apache.org/view/A-D/view/Beam/job/beam_PostCommit_Java_MavenInstall/6089/
